src/gwc/src/main/java/org/geoserver/gwc/layer/GeoServerTileLayer.java

```diff
@@ -1204,6 +1204,13 @@ public int getExpireClients(int zoomLevel) {
     /** Returns the max age of a layer group by looking for the minimum max age of its components */
     private int getGroupMaxAge(LayerGroupInfo lg) {
         int maxAge = Integer.MAX_VALUE;
+        if (isCachingEnabled(lg.getMetadata())) {
```

Comment:
So this sets an initial value for the group. However, it does not exit the method, meaning also allows for searching for a stricter value in nested layers and groups. Is this intended? I thought the group caching headers were supposed to override the single layer ones, instead as implemented, they provide just a starting pont.

Reply:
There was implemented approach pick smallest max age, I thought it will be better to follow same and pick the group max-age only if it is really shortest.
I am going to change it to pick either layer group http if exists or iterate to find min max age of all layers from group.